container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go

```diff
@@ -1138,29 +1138,22 @@ func (manager *DockerManager) FetchImage(ctx context.Context, dockerImage string
 	}
 	logrus.Tracef("Is image available locally?: %v", doesImageExistLocally)
 
-	// try and pull latest image even if image exists locally
-	if doesImageExistLocally {
-		logrus.Tracef("Image exists locally, but attempting to get latest from remote image repository.")
-		err = manager.pullImage(ctx, dockerImage)
-		if err != nil {
-			logrus.Tracef("Failed to pull Docker image '%v' from remote image repository. Going to use available local image.", dockerImage)
-		} else {
-			logrus.Tracef("Latest image successfully pulled from remote to local.")
-		}
-	} else {
+	if !doesImageExistLocally {
+		logrus.Tracef("Image doesn't exist locally, so attempting to pull it...")
 		err = manager.pullImage(ctx, dockerImage)
 		if err != nil {
-			return stacktrace.Propagate(err, "Failed to pull Docker image '%v' from remote image repository.", dockerImage)
+			return stacktrace.Propagate(err, "Failed to pull Docker image '%v' from remote image repository", dockerImage)
 		}
+		logrus.Tracef("Image successfully pulled from remote to local")
 	}
 
 	return nil
 }
 
-// [FetchLocalImage] uses the local [dockerImage] if it's available.
-// If unavailable, will attempt to fetch the latest image.
-// Returns error if local [dockerImage] is unavailable and pulling image fails.
-func (manager *DockerManager) FetchLocalImage(ctx context.Context, dockerImage string) error {
+// [FetchLatestImage] always attempts to retrieve the latest [dockerImage].
```

Comment:
this call isn't called by any function, just here so functionality exists if we want to add flag down the line